memory/src/main/scala/filodb.memory/format/vectors/DeltaDeltaVector.scala

```diff
@@ -206,6 +206,20 @@ object DeltaDeltaDataReader extends LongVectorDataReader {
     val innerIt = IntBinaryVector.simple(inner).iterate(inner, startElement)
     new DeltaDeltaIterator(innerIt, slope(vector), initValue(vector) + startElement * slope(vector).toLong)
   }
+
+  def changes(vector: BinaryVectorPtr, start: Int, end: Int): Int = {
+    if (slope(vector) == 0)
+      return 0
+    require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    var changes = 0
+    var i = start
+    while (i <= end) {
+      val current = iterate(vector, i).next
```

Comment:
This is still a problem: calling `iterate()` with each element is really expensive

Reply:
Made the changes as dicussed